cairo-contracts/packages/apps/src/transfer/components/transfer.cairo

```diff
@@ -283,19 +283,30 @@ pub mod TokenTransferComponent {
 
     #[embeddable_as(CreateIbcToken)]
     impl CreateTokenImpl<
-        TContractState, +HasComponent<TContractState>, +Drop<TContractState>,
+        TContractState,
+        +HasComponent<TContractState>,
+        +Drop<TContractState>,
+        impl Ownable: OwnableComponent::HasComponent<TContractState>,
     > of ICreateIbcToken<ComponentState<TContractState>> {
         fn create_ibc_token(
-            ref self: ComponentState<TContractState>,
-            chan_id_on_b: ChannelId,
-            base_denom: ByteArray,
+            ref self: ComponentState<TContractState>, denom: PrefixedDenom,
         ) -> ContractAddress {
-            let token = self.create_token(base_denom.clone(), 0);
-            let trace_prefix = TracePrefixTrait::new(TRANSFER_PORT_ID(), chan_id_on_b);
-            let prefixed_denom = PrefixedDenom {
-                trace_path: array![trace_prefix], base: Denom::Hosted(base_denom),
-            };
-            self.record_ibc_token(prefixed_denom, token.address);
+            let channel: ChannelContract = self.owner().into();
```

Comment:
can you please add some comments here about this `PrefixedDenom` is actually the expected `PrefixedDenom` on Starknet, not the one at counterparty.

Reply:
[530357b](https://github.com/informalsystems/ibc-starknet/pull/326/commits/530357b0e06126043834150c29fc05a88e492591)